packages/flutter/lib/src/material/dialog.dart

```diff
@@ -216,6 +217,16 @@ class AlertDialog extends StatelessWidget {
   /// from the [actions].
   final List<Widget> actions;
 
+  /// An (optional) name for the dialog used by accessibility frameworks.
+  ///
+  /// On android this would usually be a short summary of the contents. For
+  /// example - "App permissions" for a dialog containing permission
+  /// information.
```

Comment:
Maybe add that this name is usually announced by TalkBack when the route is brought on screen.

Reply:
Done